lib/src/models/utils/mapper.dart

```diff
@@ -28,6 +32,23 @@ class QMapper {
     });
   }
 
+  static QOfferings offeringsFromJson(String? jsonString) {
+    if (jsonString == null) return QOfferings(null, <QOffering>[]);
+
+    final offeringJson = Map<String, dynamic>.from(jsonDecode(jsonString));
+
+    return QOfferings.fromJson(offeringJson);
+  }
+
+  static Map<String, QEligibility> eligibilityFromJson(String? jsonString) {
```

Comment:
And here:
Just to keep the sdk in the same style I suggest using dynamic json instead of String? jsonString